### ISSUE TITLE
change in en AFF files for apostrophe usage

### ISSUE THREAD

**Issue by client9** [marcoagpinto/aoo-mozilla-en-dict#23]
Hi Marco,

The TLDR; is that I'm suggesting a change to the en_* `AFF` files that corrects the use of apostrophes (0x39) in command-line usage of `hunspell`.  I'm asking you since it appears you maintain the AOO dictionaries, and @kevina would like some confirmation that this change won't break AOO.   You can see more details here: https://github.com/kevina/wordlist/issues/122

If this is the wrong forum, please forgive me.  The process of "words" to "application dictionaries" is very confusing!  This isn't exactly an hunspell issue, and it's not an AOO issue, but yet it requires both to make this change.

Here's the short summary:

For a long time, the  hunspell CLI using the "standard" `AFF` files for `en_US` (and other `en` versions) do not handle apostrophes (ASCII 0x39) correctly.

```
$ hunspell --version
@(#) International Ispell Version 3.2.06 (but really Hunspell 1.3.4)

$ cat sample.txt
didn't  <-- 0x039
didn’t  <-- unicode
'apple'   <- 0x39
someone's  <- 0x39
someone’s  <--- unicode

# the word exists in en_US.dic
$ grep "didn't" /usr/local/share/hunspell/en_US.dic
didn't

# but hunspell thinks this is wrong
$ hunspell -l -d /usr/local/share/hunspell/en_US sample.txt
didn
didn
```

The same holds for `en_GB`, etc.

I suspect Firefox and Open Office use a different word tokenizer so they don't notice the issue.  Especially since the `en_GB` doesn't use the `WORDCHARS` stanza at all in it's AFF file.

By  adding 0x2019 `’` to the end of `WORDCHARS` makes recent versions of  hunspell (within last two years) behave correctly, IMHO.  It will be a harmless change to older versions of hunspell as well.

I'm guessing Firefox and OO don't even use the `WORDCHARS` stanza, so this change is completely safe.

In summary, my proposal is

en_US, en_CA:
change  `WORDCHARS 0123456789` to `WORDCHARS 0123456789’`

en_GB:
add `WORDCHARS ’`

I'm happy to supply a pull request if this makes the change easier. 

More details are in my original ticket https://github.com/kevina/wordlist/issues/122

Again if this is the wrong forum, please suggest how I can get in touch with the AOO team.

Thanks (and thank you for your work!)

nickg

**Comment by marcoagpinto** [comment]
Hello!

Both apostrophes work okay on Mozilla and OpenOffice (I pasted your examples text on them).

How can I test the WORDCHARS?

Tomorrow I am updating the dictionaries so I have everything ready and can't make any last moment changes (they are packed and ready to upload).

I will e-mail the AOO's mailing list asking for advice.

**Comment by client9** [comment]
Hello!  thanks.. this is not urgent at all.  It's been broken in hunspell for a long time (search for "hunspell apostrophe).

This change would only apply to command line usage of `hunspell`

Do you use linux or mac?  If so I can show you how to test.

regards,

n

**Comment by marcoagpinto** [comment]
I have VMs with Linux, mainly used to compile the software I develop.

**Comment by client9** [comment]
ok!
 
If you don't have "hunspell" installed you can do this on your VM:

```bash
git clone https://github.com/hunspell/hunspell.git
cd hunspell
./configure
make
make install
/usr/local/bin/hunspell --version
```
Ok that should say "1.3.4" somewhere.

Copy the sample file into 

```bash
hunspell -l -d DIRECTORY-OF-YOUR-AFF-DIC-FILES  sample.txt
```

The US, CA, or GB dictionary should print out `didn` a few times as being wrong.

changing the `en_XX.aff` file as follows:

| `.aff` file | change |
|-------------|----------|
| en_US, en_CA | change `WORDCHARS 0123456789` to `WORDCHARS 0123456789’`  |
| en_GB             |  add `WORDCHARS ’` anywhere in the file, but near the top is best |

Now

```bash
hunspell -l -d DIRECTORY-OF-YOUR-AFF-DIC-FILES  sample.txt
```

should print nothing, all words are correct.

Again there is no rush, and thank you for your help in maintaining these files!

regards,

nickg

**Comment by marcoagpinto** [comment]
@client9 

Hello!

Sorry for taking long.

I have added WORDCHARS ’ to the .AFF and created an OXT for you to test.

You can download it from my Dropbox:
https://dl.dropboxusercontent.com/u/30674540/dict-en-marcoagpinto-20160307.oxt

Please tell me if it fixed the issue.

Thanks!

Kind regards,
       >Marco A.G.Pinto
         ---------------------

**Comment by marcoagpinto** [comment]
@client9 
Does it work okay now? You never replied.

**Comment by client9** [comment]
it's fine, but I don't really understand who uses these files and how they are used anymore :-(

You can close this ticket if you want.  My apologies.
